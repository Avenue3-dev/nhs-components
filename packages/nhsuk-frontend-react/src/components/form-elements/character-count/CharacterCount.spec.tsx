import React from 'react';
import { it, expect } from 'vitest';
import { render } from '@testing-library/react';
import { CharacterCount } from './CharacterCount';

it('should render the CharacterCount component', () => {
  const { container } = render(
    <CharacterCount
      id="character-count"
      label="Character count"
      hint="This is a hint"
      variant="character-count"
      maxLength={10}
    />,
  );

  expect(container.querySelector('textarea')).toBeInTheDocument();
  expect(container).toMatchSnapshot();
});
