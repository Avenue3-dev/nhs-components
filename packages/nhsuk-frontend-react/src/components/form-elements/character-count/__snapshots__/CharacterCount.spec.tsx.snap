// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render the CharacterCount component 1`] = `
<div>
  <div
    class="nhsuk-character-count"
    data-maxlength="10"
    data-module="nhsuk-character-count"
  >
    <div
      class="nhsuk-form-group"
    >
      <label
        class="nhsuk-label"
        for="character-count"
        id="character-count--label"
      >
        Character count
      </label>
      <div
        class="nhsuk-hint"
        id="character-count_:r1:--hint"
      >
        This is a hint
      </div>
      <textarea
        aria-describedby="hintId"
        aria-labelledby="character-count--label"
        as="textarea"
        class="nhsuk-textarea nhsuk-js-character-count"
        id="character-count"
      />
      <div
        class="nhsuk-hint nhsuk-character-count__message"
        id="character-count-info"
      >
        You have 10 characters remaining
      </div>
    </div>
  </div>
</div>
`;
